proof/proof.go

```diff
@@ -476,12 +476,11 @@ func (p *Proof) Bytes() ([]byte, error) {
 // NOTE: This is part of the tlv.RecordProducer interface.
 func (p *Proof) Record() tlv.Record {
 	sizeFunc := func() uint64 {
-		var buf bytes.Buffer
-		err := p.Encode(&buf)
+		pBytes, err := p.Bytes()
```

Comment:
nit: use `proofBytes` instead of `pBytes`, just to unify naming a bit? Here and two more places in the diff below (`proof/proof_test.go` and `tappsbt/encode.go`).